backbone/src/main/java/org/researchstack/backbone/model/survey/SurveyItem.java

```diff
@@ -54,4 +51,30 @@ public int compare(SurveyItemType lhs, SurveyItemType rhs) {
             return 0;
         }
     }
+
+    @Override
+    public int hashCode() {
+        if (identifier == null) {
```

Comment:
One one hand, the models are probably supposed to be uniquely identifiable by the identifier, but on the other hand, client/server might get out of sync, and having equals take into account the fields could make this syncing easier.

For unit tests, I find it useful to include all the persisted fields (ones that are serialized/deserialized) for models. Calling deepCopy() should product an object that is equal. 

Whether we're comparing all persisted fields or comparing the identifier (which I assume should be globally unique) we should make that consistent across models.